go/gpg/main.go

```diff
@@ -0,0 +1,160 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"encoding/base64"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"time"
+
+	"github.com/ProtonMail/go-crypto/openpgp"
+	"github.com/ProtonMail/go-crypto/openpgp/packet"
+	"github.com/google/go-github/v54/github"
+	"golang.org/x/crypto/nacl/box"
+)
+
+var (
+	org        = flag.String("org", "", "GitHub organization name")
+	repo       = flag.String("repo", "scripts", "GitHub repo name")
+	secretName = flag.String("secret", "GPG_PRIVATE_KEY", "GitHub organization secret name")
+
+	gpgName    = flag.String("gpg-name", "opentofu", "GPG name")
+	gpgComment = flag.String("gpg-comment", "This is the key used to sign opentofu providers", "GPG comment")
+	gpgEmail   = flag.String("gpg-email", "your.email@example.com", "GPG comment")
+
+	pat string
+)
+
+func main() {
+	pat = os.Getenv("GITHUB_PAT")
+	if pat == "" {
+		panic("GITHUB_PAT environment variable not set")
+	}
+
+	flag.Parse()
+
+	config := &packet.Config{
+		DefaultCipher:          packet.CipherAES256,    // 9 for AES with 256-bit key [https://datatracker.ietf.org/doc/html/rfc4880#section-9.2]
+		DefaultCompressionAlgo: packet.CompressionZLIB, // 2 for ZLIB [https://datatracker.ietf.org/doc/html/rfc4880#section-9.3]
+		CompressionConfig: &packet.CompressionConfig{
+			Level: packet.BestCompression, // 9 for best compression [https://datatracker.ietf.org/doc/html/rfc4880#section-9.3]
+		},
+		RSABits: 4096, // 4096-bit RSA key [https://datatracker.ietf.org/doc/html/rfc4880#section-
+	}
```

Comment:
Now that you use proper const, you don't need the `9 for AES with 256-bit key` parts.

But the links I'd propose keeping, those are very useful.